utils/compress/CMakeLists.txt

```diff
@@ -10,7 +10,14 @@ add_definitions(-DNDEBUG)
 
 add_library(compress SHARED ${compress_LIB_SRCS})
 
-target_link_libraries(compress ${SNAPPY_LIBRARIES} ${LZ4_LIBRARIES})
+IF(LINK_WITH_LZ4 STREQUAL "ON")
+    MESSAGE("LINK WITH LZ4")
+    target_link_libraries(compress ${SNAPPY_LIBRARIES} ${LZ4_LIBRARIES})
+ELSE()
+    MESSAGE("LINK WITHOUT LZ4")
+    target_link_libraries(compress ${SNAPPY_LIBRARIES})
+ENDIF()
+
```

Comment:
better
```cmake
target_link_libraries(compress ${SNAPPY_LIBRARIES})
IF(LINK_WITH_LZ4)
  target_link_libraries(compress ${LZ4_LIBRARIES})
ENDIF()
```

Reply:
Thanks, fixed.

Reply:
I also meant that `LINK_WITH_LZ4` is boolean, not a string, so you don't need `STREQUAL`. But whatever you want, it's not that important.